src/com/company/Models/Users/CourseAdministrator.java

```diff
@@ -148,10 +149,98 @@ public void renameCourseModule(ArrayList<CourseModule> courseModules, CourseModu
     /**
      * @param student the student that the results slip will be about, which indicates the grades on each module.
      * @return a string that represents the results slip
-     * @throws Exception
      */
-    public String createResultsSlip(Student student) throws Exception {
-        throw new Exception("Not implemented yet");
+    public String createResultsSlip(Student student) {
+        StringBuilder resultsSlip = new StringBuilder();
+        resultsSlip
+                .append("Name: ")
+                .append(student.getFirstName())
+                .append(" ");
+        if (student.getMiddleName() != null) {
+            resultsSlip
+                .append(student.getMiddleName())
+                .append(" ");
+        }
+
+        resultsSlip.append(student.getLastName())
+                .append("     ")
+                .append("Level: ")
+                .append(student.getLevel())
+                .append("     ")
+                .append("Year: ")
+                .append(student.getYear())
+                .append("\n\n");
+
+        AsciiTable currentLevelTable = new AsciiTable();
+        currentLevelTable.addRule();
+        currentLevelTable.addRow(null, "Results for Level " + student.getLevel());
+        currentLevelTable.addRule();
+        ArrayList<CourseModuleResult> courseModulesForLevel = new ArrayList<>();
+        CourseModuleResult[] currentCourseModuleResults = student.getCurrentCourseModules();
+        ArrayList<CourseModuleResult> currentCourseModuleResultsAsArray = new ArrayList<>(Arrays.asList(currentCourseModuleResults));
+        currentCourseModuleResultsAsArray.removeAll(Collections.singleton(null));
+
+        for (CourseModuleResult currentCourseModuleResult: currentCourseModuleResultsAsArray) {
```

Comment:
Maybe check for other instances of currentCourseModuleResults in case I need to convert to an array